src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptor.cs

```diff
@@ -116,7 +117,6 @@ public string TypeName
                 }
 
                 _typeName = value;
-                IsStringProperty = string.Equals(_typeName, typeof(string).FullName, StringComparison.Ordinal);
```

Comment:
Ah I see your point. This is tough. We could potentially have the `IsStringProperty` here call through to `SanitizeFullName` couldn't we?